zap/src/main/java/org/zaproxy/zap/extension/anticsrf/AntiCsrfParam.java

```diff
@@ -87,16 +88,24 @@ protected void parse() {
             this.enabledTokensNames = new ArrayList<>(DEFAULT_TOKENS_NAMES.length);
         }
 
-        if (this.tokens.size() == 0) {
-            for (String tokenName : DEFAULT_TOKENS_NAMES) {
-                this.tokens.add(new AntiCsrfParamToken(tokenName));
-                this.enabledTokensNames.add(tokenName);
-            }
-        }
+        addMissingTokens();
 
```

Comment:
Should I call `setTokens()` here (or inside `addMissingTokens()`) so that the config is written with the updated tokens?

Reply:
Worth doing in `addMissingTokens`.